### ISSUE TITLE
Allow Annotated to wrap Final and ClassVar

### ISSUE THREAD

**Issue by GBeauregard** [python/mypy#12061]
**Bug Report**

The behavior of `Annotated` has been changed so that it is allowed to wrap `ClassVar` and `Final`. This change helps avoid friction with other users of type annotations; see https://bugs.python.org/issue46491 for context. This change has [landed in 3.11](https://github.com/python/cpython/commit/e1abffca45b60729c460e3e2ad50c8c1946cfd4e) and is in the process of being backported to [3.9](https://github.com/python/cpython/pull/30874) and [3.10](https://github.com/python/cpython/pull/30873). mypy should follow suit in allowing the wrapping in order to allow the friction lifting to happen in practice.

**To Reproduce**

```python
from typing import Annotated, ClassVar, Final

class C:
    classvar: Annotated[ClassVar[int], (2, 5)] = 4
    const: Annotated[Final[int], "metadata"] = 4
```

**Expected Behavior**

The above code should type check.

**Actual Behavior**

```console
antest/antest.py:4: error: Invalid type: ClassVar nested inside other type
antest/antest.py:5: error: Final can be only used as an outermost qualifier in a variable annotation
Found 2 errors in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: 0.931
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.9.9
- Operating system and version: Gentoo Linux stable

**Comment by hmc-cs-mdrissi** [comment]
I'd like to take this task. I'm pretty happy my comment led to this change so I'm happy to cover the pr. I'll try to do it this weekend.

**Comment by JWCS** [comment]
@hmc-cs-mdrissi Is this resolved now with [#116069](https://github.com/python/cpython/pull/116096)?

**Comment by hmc-cs-mdrissi** [comment]
No. That is related (but little different) python typing runtime change, while this issue is for type checker change. I'd forgotten about this mypy issue.
